src/saltext/github/modules/github.py

```diff
@@ -1886,4 +1897,318 @@ def _query(
+                    kwargs[param] = None
+            else:
+                kwargs[param] = _get_config_value(profile, param)
+    return kwargs
+
+
+def _format_action(rule_id=False, **kwargs):
+    """
+    Helper function to format action
+    """
+    if kwargs["ruleset_type"] == "repo":
+        action = "/".join(["repos", kwargs["owner"], kwargs["repo_name"], "rulesets"])
+    else:
+        action = "/".join(["orgs", kwargs["org_name"], "rulesets"])
+    if rule_id:
+        action = action + "/" + str(kwargs["ruleset_id"])
+    return action
+
+
+def get_ruleset(profile="github", **kwargs):
+    """
+    Get information of specific ruleset.
+
+    profile
+        The name of the profile configuration to use. Defaults to ``github``.
+    ruleset_type
+        The type of ruleset. 'repo' or 'org'.
+    ruleset_id
+        Id of the ruleset.
+    owner
+        The account owner of the repository.
+    repo_name
+        The name of the repo.
+    org_name
+        The name of the organization.
+    header_dict
+        Headers to pass to Github API.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion github.get_ruleset
+        salt myminion github.get_ruleset ruleset_id=1
+    """
+
+    kwargs = _param_dict(rule_id=True, **kwargs)
+    params = _check_params(profile, **kwargs)
+    action = _format_action(rule_id=True, **params)
+
+    try:
+        ret = _query(profile, action, header_dict=params["header_dict"])
+        if not ret.get("error"):
+            return ret["dict"]
+        else:
+            ret["comment"] = f"GitHub Response Status Code: {ret.get('error')}"
+            ret["result"] = False
+    except CommandExecutionError:
+        ret = {}
+        ret["result"] = False
+        ret["comment"] = "Error getting ruleset"
+
+    return ret
+
+
+def delete_ruleset(profile="github", **kwargs):
+    """
+    Delete ruleset
+
+    profile
+        The name of the profile configuration to use. Defaults to ``github``.
+    ruleset_type
+        The type of ruleset. 'repo' or 'org'.
+    owner
+        The account owner of the repository.
+    repo_name
+        The name of the repo.
+    ruleset_id
+        Id of the ruleset.
+    org_name
+        The name of the organization.
+    header_dict
+        Headers to pass to Github API.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion github.delete_ruleset
+        salt myminion github.delete_ruleset ruleset_id=1
+    """
+    kwargs = _param_dict(rule_id=True, **kwargs)
+    params = _check_params(profile, **kwargs)
+    action = _format_action(rule_id=True, **params)
+
+    try:
+        ret = _query(profile, action, method="DELETE", header_dict=params["header_dict"])
+        if not ret.get("error"):
+            ret["comment"] = f"ruleset {params['ruleset_id']} successfully deleted"
+        else:
+            ret["comment"] = f"GitHub Response Status Code: {ret.get('error')}"
+            ret["result"] = False
```

Comment:
**suggestion (non-blocking):** it's more readable to check the positive instead of the negative

```suggestion
        if ret.get("error"):
            ret["comment"] = f"GitHub Response Status Code: {ret.get('error')}"
            ret["result"] = False
        else:
            ret["comment"] = f"ruleset {params['ruleset_id']} successfully deleted"
```